libstuff/STCPServer.cpp

```diff
@@ -55,14 +55,8 @@ STCPManager::Socket* STCPServer::acceptSocket(Port*& portOut) {
         if (s > 0) {
             // Received a socket, wrap
             SDEBUG("Accepting socket from '" << addr << "' on port '" << port.host << "'");
-            socket = new Socket;
-            socket->s = s;
+            socket = new Socket(s, STCP_CONNECTED);
```

Comment:
I found a different bug, where we'd accept a socket and then not set it's `lastRecvTime`, which would be `0` (or any random number, but often 0), which would result in the socket being in a state where it had timed out already, and cause it to immediately try and reconnect.

This only happened in a certain situation that I'd have to go back and look through the commits to find, but I decided to fix it by improving the socket constructor to initialize everything, and thought I might as well pass in essential state as well.